src/op_handlers/context.rs

```diff
@@ -0,0 +1,75 @@
+use u256::U256;
+use zkevm_opcode_defs::ethereum_types::Address;
+use zkevm_opcode_defs::VmMetaParameters;
+
+use crate::address_operands::{address_operands_read, address_operands_store};
+use crate::value::TaggedValue;
+use crate::{opcode::Opcode, state::VMState};
+
+// consider moving this function to a utils crate
+// taken from matter-labs zk evm implementation
+fn address_to_u256(address: &Address) -> U256 {
+    let mut buffer = [0u8; 32];
+    buffer[12..].copy_from_slice(&address.as_fixed_bytes()[..]);
+
+    U256::from_big_endian(&buffer)
+}
+
+pub fn _this(vm: &mut VMState, opcode: &Opcode) {
+    let res = TaggedValue::new_raw_integer(address_to_u256(&vm.current_frame().address));
+    address_operands_store(vm, opcode, res);
+}
+
+pub fn _caller(vm: &mut VMState, opcode: &Opcode) {
+    let res = TaggedValue::new_raw_integer(address_to_u256(&vm.current_frame().caller));
+    address_operands_store(vm, opcode, res);
+}
+
+pub fn _code_address(vm: &mut VMState, opcode: &Opcode) {
+    let res = TaggedValue::new_raw_integer(address_to_u256(&vm.current_frame().code_address));
+    address_operands_store(vm, opcode, res);
+}
```

Comment:
I infer, from the names, that these should be creating pointers rather than integers.

Reply:
I don't think they are, since `TaggedValue::new_pointer` actually indicates that a Fat Pointer is being stored, which is not the case, if it leads to confusion we can change it to `TaggedValue::new_fat_pointer`

Reply:
Yes, this is due to a mistake, it should be named `context_ergs_left.zasm` instead of `context_code_ergs_left.zasm`

Reply:
I think that would be sensible.